openedx/core/lib/edx_api_utils.py

```diff
@@ -59,7 +59,17 @@ def get_edx_api_data(api_config, resource, api, resource_id=None, querystring=No
         response = endpoint(resource_id).get(**querystring)
 
         if resource_id is not None:
-            results = response
+            if fields:
+                results = {}
+                for field in fields:
+                    try:
+                        results[field] = response[fields]
```

Comment:
Might be safer to use `get`, in case the requested field isn't in the response. You could also make an argument for allowing it to throw an error in that case though.

Reply:
I figure that catching the exception and logging the error would be helpful for identifying problems relating to the field not being in the response.